### ISSUE TITLE
choice grid inaccessible to screen readers

### ISSUE THREAD

**Issue by fnetX** [cryptpad/cryptpad#1601]
### Description

Offtopic / Appreciation: I appreciate cryptpad and the efforts you make, also with regards to usability improvements. I use Cryptpad often for surveys on usability.

---

I just wanted to test how (easy) navigation of a shared cryptpad poll / form can be done with a screen reader. Partially, because I am interested (since I share the polls regularly, I want to be sure people can make use of them to answer questions and not to introduce a bias because they are impossible to fill for people with limited vision). And secondly, because I am currently also trying to improve accessibility on similar option grids for [Forgejo](https://forgejo.org).

While most of my test form worked acceptable, this part of it was barely usable. Thank you for improving accessibility.

### Steps to reproduce

1. launch a screen reader, it can be demonstrated with the libre orca
2. navigate to a public form that contains a choice grid
3. confirm that the text is read
4. close your eyes! Really!
5. try to navigate back and forth between the options
6. **issue** options are not announced as radio buttons, but as "blank" for me
7. **issue** the options are not labelled, it is hard to know what the buttons mean
8. **Issue:** navigating straight up to get the header of the column is not possible

### Expected behavior

- The option grid behaves like a table, so that
  - Navigating up to the header is possible
  - reading a whole line is possible (announces the description, the state of the checkboxes and the meaning from the table header

### Which operating system are you using?

Linux/BSD/UNIX

### What browsers are you seeing the problem on?

Mozilla Firefox

### Extensions

not relevant

### Version

2024.6.1

**Comment by fnetX** [comment]
I don't know about the Cryptpad UI, but I usually recommend to use semantic HTML. It gives you accessibility and most features out of the box. If the table was an actual table and not a structure of divs, it would be much more accessible. If the option buttons were actually option buttons and not spans, they would also be more accessible.
